x/programs/cmd/simulator/src/lib.rs

```diff
@@ -114,6 +113,7 @@ pub(crate) struct SimulatorTestContext {
 #[derive(Clone, Debug, PartialEq)]
 pub enum Param {
     U64(u64),
+    U32(u32),
```

Comment:
This is true, but we still need strong typing on the Rust side. Otherwise we could as well borsh serialize everything but that gives fewer guarantees about the type and is harder to debug.

Reply:
To clarify, this PR https://github.com/ava-labs/hypersdk/pull/1099 makes it now unnecessary to add the types on the Go side as well, since any unregistered type which doesn't need special treatment is just passed as "bytes".

Reply:
I removed the use of this type since it was not *necessary* but it might make sense to introduce it back later.